stablehlo/dialect/TypeInference.cpp

```diff
@@ -3563,6 +3605,60 @@ LogicalResult verifyConvolutionOp(
                              "is incompatible with return type of operation ",
                              shapedResultType, "");
 
+  // If the operation uses quantized tensors:
+  if (!noneQuantized<quant::QuantizedType>({lhsType, rhsType, resultType})) {
+    Type lhsElementType = getElementTypeOrSelf(lhsType);
```

Comment:
instead of 
Type -> ElementType -> using `isa<>` checks -> cast to QuantizedTypes.
I think, 
Type -> cast to QuantizedTypes and use null checks seems simpler. also, less verbose vs need to maintain "ElementType" in name and it aligns with the other verifier implementations (example: reshapeOp). 
what do you think? This may be not applicable once you move checks to dedicated API.

Reply:
I prefer having *ElementType in name and explicit type check for clarity. After applying the new structure as you suggested, I think it doesn't look too verbose. How do you think?